core/common-util/src/main/java/uk/gov/gchq/gaffer/commonutil/iterable/LimitedCloseableIterator.java

```diff
@@ -52,12 +64,18 @@ public void close() {
 
     @Override
     public boolean hasNext() {
-        boolean hasNext = (null == end || index < end) && iterator.hasNext();
-        if (!hasNext) {
+        boolean hasNext = (null == end || index < end);
+        boolean iteratorHasNext = iterator.hasNext();
+
+        if (!hasNext && iteratorHasNext && !truncate) {
+            throw new LimitExceededException("Limit exceeded - not all data could be shown");
```

Comment:
Please could you add the 'end' field to the error message so users know what limit was hit.